### PR TITLE
[ios] do not loop inaudible audio for silent switch hack

### PR DESCRIPTION
**Pull request by nealmanaktola**
~~it seems like looping audio doesn't seem to be necessary~~

~~when the audio track is not looped, I don't see a bunch of worker threads being spawned to play the audio~~

Instead of looping the inaudible audio, the audio is played every 2 seconds and when the app is foregrounded. I couldn't find a way to know when the WkWebView audio session category changes, as it would be better to play the audio element only when needed.

### CODE REVIEW

File: apple/RNCWebView.m [hunk 1]
```diff
@@ -1365,19 +1365,32 @@ - (void)evaluateJS:(NSString *)js
...
     NSString *scr;
+
+    int timeIntervalForInaudibleAudioMs = 2000;
     if (initialSetup) {
-        scr = [NSString stringWithFormat:@"var s=new Audio('%@');s.id='wkwebviewAudio';s.controls=false;s.loop=true;s.play();document.body.appendChild(s);true", mp3Str];
+        scr = [NSString stringWithFormat:@"var s=new Audio('%@');s.id='wkwebviewAudio';s.controls=false;s.play();document.body.appendChild(s);var playInterval = setInterval(() => s.play(), %d);true", mp3Str, timeIntervalForInaudibleAudioMs];
     } else {
-        scr = [NSString stringWithFormat:@"document.getElementById('wkwebviewAudio').muted=false", mp3Str];
+        // When the app is foregrounded, the audio element is played again and the interval is restarted.
+        scr = [NSString stringWithFormat:@"var audio=document.getElementById('wkwebviewAudio');audio.muted=false;audio.play();var playInterval = setInterval(() => s.play(), %d);true", timeIntervalForInaudibleAudioMs];
```
Comment (apkumar) [inline apple/RNCWebView.m:1385]: wouldn't we need to stop the old interval?

Comment (apkumar) [reply apple/RNCWebView.m:1385]: i see.

and with this 'play on an interval' method, we don't see the same activity as we do when we are looping?

Comment (nealmanaktola) [reply apple/RNCWebView.m:1385]: the activity is much less
before:
this is  for one of the mediaserverd threads
Screen Shot 2022-12-20 at 1 37 31 PM

after:

Screen Shot 2022-12-20 at 1 32 26 PM

i am making the assumption that these threads are equivalent in the sense that both of them are responsible for playing the audio. in the second screenshot, i validated that there is some cpu activity every 2s. i did not see a thread that resembled the first screen shot (which had loop=true)

Comment (nealmanaktola) [reply apple/RNCWebView.m:1385]: I realized the second screenshot didn't have the timescale for you to verify that it's every 2 s
Screen Shot 2022-12-20 at 1 32 31 PM

**Pull request merged**